packages/ERTP/src/issuer.js

```diff
@@ -154,75 +158,95 @@ function produceIssuer(allegedName, mathHelpersName = 'nat') {
       purseLedger.init(purse, amountMath.getEmpty());
       return purse;
     },
-    isLive: paymentLedger.has,
-    getAmountOf: paymentLedger.get,
+
+    isLive: paymentP => {
+      return Promise.resolve(paymentP).then(payment => {
+        return paymentLedger.has(payment);
+      });
+    },
+    getAmountOf: paymentP => {
+      return Promise.resolve(paymentP).then(payment => {
+        let result;
+        try {
+          result = paymentLedger.get(payment);
+        } catch (e) {
+          throw new Error(`payment not found: ${allegedName}`, e);
+        }
+        return result;
+      });
+    },
     burn: (paymentP, optAmount = undefined) => {
-      const payment = E.unwrap(paymentP);
-      const paymentBalance = paymentLedger.get(payment);
-      assertAmountEqual(paymentBalance, optAmount);
-      // Commit point.
-      paymentLedger.delete(payment);
-      return paymentBalance;
+      return Promise.resolve(paymentP).then(payment => {
+        const paymentBalance = paymentLedger.get(payment);
+        assertAmountEqual(paymentBalance, optAmount);
+        // Commit point.
+        paymentLedger.delete(payment);
+        return paymentBalance;
+      });
     },
     claim: (paymentP, optAmount = undefined) => {
-      const srcPayment = E.unwrap(paymentP);
-      const srcPaymentBalance = paymentLedger.get(srcPayment);
-      assertAmountEqual(srcPaymentBalance, optAmount);
-      // Commit point.
-      const [payment] = reallocate(
-        harden({
-          payments: [srcPayment],
-          newPaymentBalances: [srcPaymentBalance],
-        }),
-      );
-      return payment;
+      return Promise.resolve(paymentP).then(srcPayment => {
+        const srcPaymentBalance = paymentLedger.get(srcPayment);
+        assertAmountEqual(srcPaymentBalance, optAmount);
+        // Commit point.
+        const [payment] = reallocate(
+          harden({
+            payments: [srcPayment],
+            newPaymentBalances: [srcPaymentBalance],
+          }),
+        );
+        return payment;
+      });
     },
     // Payments in `fromPaymentsPArray` must be distinct. Alias
     // checking is delegated to the `reallocate` function.
     combine: (fromPaymentsPArray, optTotalAmount = undefined) => {
-      const fromPaymentsArray = fromPaymentsPArray.map(E.unwrap);
-      const totalPaymentsBalance = fromPaymentsArray
-        .map(paymentLedger.get)
-        .reduce(add, empty);
-      assertAmountEqual(totalPaymentsBalance, optTotalAmount);
-      // Commit point.
-      const [payment] = reallocate(
-        harden({
-          payments: fromPaymentsArray,
-          newPaymentBalances: [totalPaymentsBalance],
-        }),
-      );
-      return payment;
+      return Promise.resolve(fromPaymentsPArray).then(fromPaymentsArray => {
```

Comment:
BUG

`fromPaymentsPArray` is not a promise for an array of payments. It is an array of promises for payments, which is why the `P` appears where it is in the name. The old code handled this with `.map`. For this code, just replace `resolve` with `all`.

Reply:
I added a test that supplies an array of promises. (from `claim()`).